br/pkg/lightning/backend/local/local.go

```diff
@@ -266,11 +266,30 @@ func (g *targetInfoGetter) CheckRequirements(ctx context.Context, checkCtx *back
 	if err := tikv.CheckTiKVVersion(ctx, g.tls, g.pdAddr, localMinTiKVVersion, localMaxTiKVVersion); err != nil {
 		return err
 	}
+	clusterId, err := FetchClusterID(ctx, db)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err := tikv.CheckTiDBDestination(ctx, g.tls, g.pdAddr, clusterId); err != nil {
+		return err
+	}
 
 	serverInfo := version.ParseServerInfo(versionStr)
 	return checkTiFlashVersion(ctx, g.targetDBGlue, checkCtx, *serverInfo.ServerVersion)
 }
 
+// FetchClusterID gets the cluster id from tidb server using query
+func FetchClusterID(ctx context.Context, db utils.QueryExecutor) (string, error) {
+	var clusterId string
+	const queryTiDB = "select substring(type,8) from METRICS_SCHEMA.PD_CLUSTER_METADATA limit 1;"
+	tidbRow := db.QueryRowContext(ctx, queryTiDB)
```

Comment:
one potential issue is that this won't work if PD has not enabled their Prometheus port.

Reply:
i changed the comparing way of cluster id as description.